File: backend/models/indexing.py
"""
Indexing status data model.

This module defines models for indexing status tracking.
"""

from typing import Literal, List
from pydantic import BaseModel


class BookIndexingStatus(BaseModel):
    """Model representing indexing status of a single book."""

    book_id: str
    status: Literal["pending", "indexing", "completed", "failed"]
    progress: int  # 0-100


class IndexingStatusResponse(BaseModel):
    """Model for overall indexing status response."""

    books_indexing: List[BookIndexingStatus]
    overall_status: Literal["pending", "indexing", "completed", "failed"]
    total_books: int
    indexed_books: int
    in_progress: bool

File: frontend/src/types/indexing.ts
/**
 * Type definitions for Indexing status data structures.
 */

export type IndexingStatus = 'pending' | 'indexing' | 'completed' | 'failed';

export interface BookIndexingStatus {
  book_id: string;
  status: IndexingStatus;
  progress: number;
}

export interface IndexingStatusResponse {
  books_indexing: BookIndexingStatus[];
  overall_status: IndexingStatus;
}
